### ISSUE TITLE
[a11y]: Inline Notification are not announced by screen reader tools

### ISSUE THREAD

**Issue by RakshaFauzdar** [carbon-design-system/carbon#22720]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.99.0

### React version

18.3.0

### Automated testing tool and ruleset

JAWS and  https://www.ibm.com/able/requirements/requirements/#4_1_3

### Assistive technology

JAWs

### Description

Inline Notification are not announced by screen reader tools.
Although I have tried to add aria-live="assertive" and role ="status" as well . still it is not annouced by JAWS screen reader tool.

### WCAG 2.1 Violation

https://www.ibm.com/able/requirements/requirements/#4_1_3

### Reproduction/example

https://stackblitz.com/edit/github-eyjq5bem?file=README.md&preset=node=

### Steps to reproduce

Implement any notification on any button click .
Use Tab key and click on button and then screen reader should announce it .

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
